File: go/reducing-dishes.go
func maxSatisfaction(satisfaction []int) int {
	sort.Ints(satisfaction)
	currentSum := 0
	currentSatisfaction := 0
	maximalSatisfactiom := 0
	for i := len(satisfaction) - 1; 0 <= i; i-- {
		currentSum += satisfaction[i]
		currentSatisfaction += currentSum
		if maximalSatisfactiom < currentSatisfaction {
			maximalSatisfactiom = currentSatisfaction
		}
	}
	return maximalSatisfactiom
}

File: python/reducing-dishes.py
class Solution:
    def maxSatisfaction(self, satisfaction: List[int]) -> int:
        satisfaction.sort()
        current_sum = 0
        max_satisfaction = 0
        current_satisfaction = 0
        while satisfaction:
            current_sum += satisfaction.pop()
            current_satisfaction += current_sum
            max_satisfaction = max(max_satisfaction, current_satisfaction)
        return max_satisfaction
